modules/ding_user_frontend/ding_user_frontend.module

```diff
@@ -307,6 +307,14 @@ function ding_user_frontend_url_inbound_alter(&$path, $original_path, $path_lang
       if (isset($matches[1])) {
         $path .= '/' . $matches[1];
       }
+    } else {
+      // Don't cache this page so we don't get any weird caching problems.
+      drupal_page_is_cacheable(FALSE);
+
+      // If the user is not logged in we send them to login page.
+      $options = ['query' => ['destination' => $path]];
+      $login_url = ding_provider_invoke( 'openplatform_token', 'login_url');
+      drupal_goto($login_url, $options);
```

Comment:
I do not think this is the right way to address this:

1. It creates a dependency between the `ding_user_frontend` module and the `openplatform_token` provider.
2. The `openplatform_token` provider is used to determine the login url in a context where there is no clear line between login and Open Platform.

Drupal does not provide a hook to determine login urls but it does have a variable `site_403` which contains the path users will be shown in this context.

As an alternate solution to this problem I suggest adding an update hook to the `ding_adgangsplatformen` module which sets `site_403` to `DING_ADGANGSPLATFORMEN_LOGIN_URL`.

A similar approach is used in other Drupal modules such as [`r4032login`](https://git.drupalcode.org/project/r4032login/-/blob/7.x-1.x/r4032login.install) and [`logintoboggan`](https://git.drupalcode.org/project/logintoboggan/-/blob/7.x-1.x/logintoboggan.admin.inc#L203).

This will effectively redirect all 403 responses to Adgangsplatformen login page but I do not think that should be a problem.

Reply:
I dont think the proposed solution works. The most common user case is the user accessing there status pages. E.g /user/me/status-reservations . But that returns a 404 Not Found for users who are not logged in. So the code which uses the site_403 variable is never runned.

Reply:
On possible solution would be to create a hook for this situation and then implement it in ding_adgangsplatformen. You get the same result as this code but without  creating a depency between ding_user_frontend and the ding_adgangsplatformen.

Reply:
:+1: I see your point @arniloftsson and I think your alternate solution sounds like a good idea. If you also still think it is a reasonable solution then let's go with that.